src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,37 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+
+    public String value;
+    protected boolean isPrivate;
+
+    public  Contact( String contact, boolean isPrivate) {
+        this.isPrivate = isPrivate;
+        String trimmedContact = contact.trim();
+        this.value = trimmedContact;
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
```

Comment:
This is fine, but note that now an Email and Address having the same value will be considered equal even if they are of different types. You should override this method in child classes if possible.